tutorials/auto_scheduler/tune_network_arm.py

```diff
@@ -0,0 +1,420 @@
+#     |   15 |        0.149 |         689.13 |    128 |
+#     |   16 |        0.155 |         664.80 |    192 |
+#     |   17 |        0.151 |         340.64 |     64 |
+#     |   18 |        0.176 |         597.55 |    128 |
+#     |   19 |        0.220 |        1054.37 |    192 |
+#     |   20 |        0.150 |         686.01 |    128 |
+#     |   21 |        0.159 |         650.88 |    128 |
+#     |   22 |        0.073 |         358.19 |     64 |
+#     |   23 |        0.031 |          70.63 |     64 |
+#     |   24 |        0.251 |         947.73 |    128 |
+#     |   25 |        0.157 |         652.47 |    128 |
+#     |   26 |        0.215 |         954.84 |    128 |
+#     |   27 |        0.237 |         868.92 |    128 |
+#     |   28 |        0.266 |         774.06 |    128 |
+#     -------------------------------------------------
+#     Estimated total latency: 10.016 ms      Trials: 3992    Used time : 1131 s      Next ID: 15
+#
+#   This table lists the latency and (estimated) speed of all tasks.
+#   It also lists the allocation of measurement trials for all tasks.
+#   The last line prints the total weighted latency of these tasks,
+#   which can be a rough estimation of the end-to-end execution time
+#   of the network.
+#   The last line also prints the total number of measurement trials,
+#   total time spent on auto-tuning and the id of the next task to tune.
+#
+#   There will also be some "dmlc::Error"s errors, because the
+#   auto-scheduler will try some invalid schedules.
+#   You can safely ignore them if the tuning can continue, because these
+#   errors are isolated from the main process.
+#
+
+######################################################################
+# .. note:: Terminate the tuning earlier
+#
+#   You can terminate the tuning earlier by forcibly killing this process.
+#   As long as you get at least one valid schedule for each task in the log file,
+#   you should be able to do the compilation (the secion below).
+#
+
+
+#################################################################
+# Compile and Evaluate
+# --------------------
+# After auto-tuning, we can compile the network with the best schedules we found.
+# All measurement records are dumped into the log file during auto-tuning,
+# so we can read the log file and load the best schedules.
+
+# Compile with the history best
+print("Compile...")
+with auto_scheduler.ApplyHistoryBest(log_file):
+    with tvm.transform.PassContext(opt_level=3, config={"relay.backend.use_auto_scheduler": True}):
+        lib = relay.build(mod, target=target, params=params)
+
+# Export library
+tmp = tempdir()
+if use_android:
+    from tvm.contrib import ndk
+
+    filename = "net.so"
+    lib.export_library(tmp.relpath(filename), ndk.create_shared)
+else:
+    filename = "net.tar"
+    lib.export_library(tmp.relpath(filename))
+
+# Upload module to device
+print("Upload...")
+remote = autotvm.measure.request_remote(device_key, "0.0.0.0", 9191, timeout=10000)
```

Comment:
- It's a bit weird to see `autotvm` here. Maybe we should lift the common modules up to the top-level so that it can serve autotvm, auto_schedueler, and deployments.
- I didn't find the corresponding code of launching an RPC server at `0.0.0.0:9191` in this tutorial. How does this line work on the CI?

Reply:
Thank you @comaniac for the input! Great point with lifting common modules up from autotvm here. @merrymercy do you have opinions on this?

And great catch on requesting a remote in the tutorial, this code won't run in CI. I've commented it out so it doesn't run in CI.

Reply:
use this https://github.com/apache/tvm/blob/6787d7494f8815bce9523906935169f6385b9d93/python/tvm/auto_scheduler/utils.py#L242

Reply:
Thanks @merrymercy I am now using this request_remote, there is no dependence of the autotvm lib in this tutorial anymore!